src/components/Feed/Reactions.tsx

```diff
@@ -0,0 +1,91 @@
+import { Bomb, Fire, HandsPraying, Heart, Horse, Spinner } from "@phosphor-icons/react"
+import { useQuery } from "@tanstack/react-query";
+import { motion } from "framer-motion";
+import { forwardRef, useImperativeHandle, useRef, useState } from "react";
+import { api_reactions } from "../../shared/api";
+
+const reactionIcons = {
+  like: {
+    Icon: Heart,
+  },
+  unicorn: {
+    Icon: Horse,
+  },
+  exploding_head: {
+    Icon: Bomb,
+  },
+  raised_hands: {
+    Icon: HandsPraying,
+  },
+  fire: {
+    Icon: Fire,
+  },
+};
+
+export type ReactionsHandler = {
+    toggle: (articleId: number) => void;
+};
+
+export const Reactions = forwardRef<ReactionsHandler>((_, ref) => {
+  const [articleId, setArticleId] = useState<number | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const backdropRef = useRef<HTMLDivElement>(null);
+
+  const { data: reactions, isPending } = useQuery({
+    queryKey: ["reactions", articleId],
+    enabled: !!articleId,
+    queryFn: async () => {
+      const params = new URLSearchParams({
+        article_id: articleId?.toString() ?? "",
+      });
+      return (await api_reactions.get("/reactions?" + params)).data as Reactions;
+    },
+  });
+
+  useImperativeHandle(ref, () => ({
+    toggle: (articleId: number) => {
+      setArticleId(articleId);
+      setIsOpen(!isOpen);
+    },
+  }));
+
+  return (
+    <>
+      <div
+        ref={backdropRef}
+        className={`fixed inset-0 w-full h-full z-50 ${isOpen ? "" : "pointer-events-none"}`}
```

Comment:
I believe we can hide it entirely when not used
```suggestion
        className={`fixed inset-0 w-full h-full z-50 ${isOpen ? "" : "hidden"}`}
```